File: backend/src/scripts/import-places.ts
import { db } from '../db'
import { trips, days, items } from '../db/schema'
import * as fs from 'fs'
import * as path from 'path'

interface Place {
  type: string
  city: string
  name: string
  lngLat: number[]
  distance: string | null
  duration: string | null
  path: string | null
  description: string | null
  ticket: string | null
  address: string
  phone: string
  openingHours: string
  rating: string
  category: string
  notes: string
}

async function importPlaces() {
  const placesPath = path.resolve(__dirname, '../../../places1.json')
  const raw = fs.readFileSync(placesPath, 'utf-8')
  const places: Place[] = JSON.parse(raw)

  // Group by city
  const grouped = places.reduce<Record<string, Place[]>>((acc, place) => {
    if (!acc[place.city]) acc[place.city] = []
    acc[place.city].push(place)
    return acc
  }, {})

  for (const [city, cityPlaces] of Object.entries(grouped)) {
    console.log(`Importing ${cityPlaces.length} places for ${city}...`)

    await db.transaction(async (tx) => {
      const [trip] = await tx.insert(trips).values({
        title: city
      }).returning()

      const [day] = await tx.insert(days).values({
        dayIndex: 0,
        tripId: trip.id
      }).returning()

      await tx.insert(items).values(
        cityPlaces.map((p) => ({
          type: 'place',
          name: p.name,
          lngLat: p.lngLat,
          distance: p.distance,
          duration: p.duration,
          path: p.path,
          description: p.description,
          ticket: p.ticket,
          address: p.address,
          phone: p.phone,
          openingHours: p.openingHours,
          rating: p.rating,
          category: p.category,
          notes: p.notes,
          dayId: day.id
        }))
      )
    })

    console.log(`  Done.`)
  }

  console.log('Import completed.')
  process.exit(0)
}

importPlaces().catch((err) => {
  console.error('Import failed:', err)
  process.exit(1)
})

File: backend-py/scripts/import_places.py
#!/usr/bin/env python3
import asyncio
import json
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent.parent))

from app.database import async_session_factory
from app.models import Day, Item, Trip


async def import_places() -> None:
    places_path = Path(__file__).parent.parent.parent / "places1.json"
    if not places_path.exists():
        print(f"Places file not found: {places_path}")
        sys.exit(1)

    with open(places_path, "r", encoding="utf-8") as f:
        places = json.load(f)

    grouped: dict[str, list[dict]] = {}
    for place in places:
        city = place.get("city", "Unknown")
        if city not in grouped:
            grouped[city] = []
        grouped[city].append(place)

    async with async_session_factory() as session:
        for city, city_places in grouped.items():
            print(f"Importing {len(city_places)} places for {city}...")

            trip = Trip(title=city)
            session.add(trip)
            await session.flush()

            day = Day(dayIndex=0, tripId=trip.id)
            session.add(day)
            await session.flush()

            for p in city_places:
                item = Item(
                    type="place",
                    name=p.get("name", ""),
                    lngLat=json.dumps(p.get("lngLat", [])),
                    distance=p.get("distance"),
                    duration=p.get("duration"),
                    path=json.dumps(p.get("path")) if p.get("path") else None,
                    description=p.get("description"),
                    ticket=p.get("ticket"),
                    address=p.get("address", ""),
                    phone=p.get("phone", ""),
                    openingHours=p.get("openingHours", ""),
                    rating=p.get("rating", ""),
                    category=p.get("category", ""),
                    notes=p.get("notes", ""),
                    dayId=day.id,
                )
                session.add(item)

            print("  Done.")

        await session.commit()

    print("Import completed.")


if __name__ == "__main__":
    asyncio.run(import_places())
